payjoin-client/src/main.rs

```diff
@@ -174,28 +176,45 @@ impl App {
         let _to_broadcast_in_failure_case = proposal.get_transaction_to_schedule_broadcast();
 
         // The network is used for checks later
-        let network = match self.bitcoind.get_blockchain_info()?.chain.as_str() {
+        let network = match self
+            .bitcoind
+            .get_blockchain_info()
+            .map_err(|e| Error::Server(e.into()))?
+            .chain
+            .as_str()
+        {
             "main" => bitcoin::Network::Bitcoin,
             "test" => bitcoin::Network::Testnet,
             "regtest" => bitcoin::Network::Regtest,
-            _ => return Err(ReceiveError::Other(anyhow!("Unknown network"))),
+            _ => return Err(Error::Server(anyhow!("Unknown network").into())),
         };
 
         // Receive Check 1: Can Broadcast
         let proposal = proposal.check_can_broadcast(|tx| {
-            self.bitcoind
-                .test_mempool_accept(&[bitcoin::consensus::encode::serialize(&tx).to_hex()])
-                .unwrap()
-                .first()
-                .unwrap()
-                .allowed
+            let raw_tx = bitcoin::consensus::encode::serialize(&tx).to_hex();
+            let mempool_results = self
+                .bitcoind
+                .test_mempool_accept(&[raw_tx])
+                .map_err(|e| Error::Server(e.into()))?;
+            match mempool_results.first() {
+                Some(result) => Ok(result.allowed),
+                None => Err(Error::Server(
+                    anyhow!("No mempool results returned on broadcast check").into(),
+                )),
+            }
         })?;
         log::trace!("check1");
 
         // Receive Check 2: receiver can't sign for proposal inputs
         let proposal = proposal.check_inputs_not_owned(|input| {
-            let address = bitcoin::Address::from_script(input, network).unwrap();
-            self.bitcoind.get_address_info(&address).unwrap().is_mine.unwrap()
+            if let Ok(address) = bitcoin::Address::from_script(input, network) {
```

Comment:
There is discrepancy between how errors are handled in check code. Here, the from_script error is discarded, since the PSBT will be invalid if the script is invalid and won't be able to be broadcast